### PR TITLE
fix(cli/fetch): set null body for null-body status

### LINKED ISSUE #5974: TypeError in fetch with HTTP 204 response
**To reproduce:**
```
$ deno eval 'console.log((await fetch("https://httpbin.org/status/204")).status)'
```

**Actual output:**
```
error: Uncaught TypeError: Response with null body status cannot have body
    at new Response ($deno$/web/fetch.ts:49:13)
    at fetch ($deno$/web/fetch.ts:329:22)
    at async file:///.../__$deno$eval.ts:1:14
```
version string:
```
$ deno --version
deno 1.0.3
v8 8.4.300
typescript 3.9.2
```

**Expected output**:
```
204
```
For example, this is the current behavior when 200 is used in place of 204:
```
$ deno eval 'console.log((await fetch("https://httpbin.org/status/200")).status)'
200
```

### PR DESCRIPTION
**Pull request by marcosc90**
Fixes #5974

The [spec](https://fetch.spec.whatwg.org/#ref-for-null-body-status) says the following:

> If response is not a network error and either request’s method is `HEAD` or `CONNECT`, or internalResponse’s status is a null body status, set internalResponse’s body to null and disregard any enqueuing toward it (if any).

I followed the spec in this PR. if `status` is any of `204, 205, 304`, `res.body` will be `null`. 
Both Firefox & Chrome let you read the response body for a `204` (Even if it shouldn't have a body). 

When using `Response` constructor Deno, Firefox & Chrome works the same way:

```
new Request('body', { status: 204 }); // TypeError
```

### CODE REVIEW

File: tools/http_server.py [hunk 1]
```diff
@@ -194,8 +194,9 @@ def do_GET(self):
     def do_POST(self):
         # Simple echo server for request reflection
         if "echo_server" in self.path:
+            status = int(self.headers.getheader('x-status', "200"))
```
Comment (marcosc90) [inline tools/http_server.py:197]: This is working but I don't know python, so if there's a better way please let me know :)

Comment (ry) [reply tools/http_server.py:197]: that works

**Comment by ry** [comment]
Oops we have a merge conflict. @marcosc90 please merge with master

**Comment by marcosc90** [comment]
Thanks @ry and already fixed merge conflicts.

**Pull request merged**